dsl_parser/import_resolver/resolver_with_catalog_support.py

```diff
@@ -0,0 +1,246 @@
+    def _is_blueprint_url(import_url):
+        return import_url.startswith(BLUEPRINT_PREFIX)
+
+    def fetch_import(self, import_url):
+        if self.mappings:
+            import_url = self._rewrite_from_mappings(import_url)
+        if self._is_blueprint_url(import_url):
+            return self._fetch_blueprint_import(import_url)
+        elif self._is_plugin_url(import_url):
+            return self._fetch_plugin_import(import_url)
+        return super(ResolverWithCatalogSupport, self).fetch_import(import_url)
+
+    def _rewrite_from_mappings(self, import_url):
+        for plugin_name, mapping in self.mappings.items():
+            if import_url == mapping.get('import_url'):
+                return 'plugin:{0}?version={1}'.format(
+                    plugin_name,
+                    mapping.get('version')
+                )
+        return import_url
+
+    def _fetch_plugin_import(self, import_url):
+        import_url = self._resolve_plugin_yaml_url(import_url)
+        return super(ResolverWithCatalogSupport, self).fetch_import(import_url)
+
+    @staticmethod
+    def _make_plugin_filters(plugin_spec, version_constraints, mappings):
+        """Parse the plugin spec to a dict of filters for the sql query
+
+        >>> _make_plugin_filters('cloudify-openstack-plugin')
+        {'package_name': 'cloudify-openstack-plugin'}
+        >>> _make_plugin_filters('cool?version=1.0.2')
+        {'package_name': 'cool', 'package_version': '1.0.2'}
+        >>> _make_plugin_filters('cool?version=1.0.2&distribution=centos')
+        {'package_name': 'cool', 'package_version': '1.0.2',
+         'distribution': 'centos'}
+        """
+        filter_renames = {'version': 'package_version',
+                          'distribution': 'distribution'}
+        name, _, params = plugin_spec.partition('?')
+        filters = {}
+        for filter_name, value in parse_qs(params).items():
+            renamed = filter_renames.get(filter_name)
+            if renamed is None:
+                raise InvalidBlueprintImport(
+                    'Error parsing spec for plugin {0}: invalid parameter {1}'
+                    .format(name, filter_name))
+            filters[renamed] = value
+
+        # In case a mapping for package_version was provided, use it above
+        # all other specs.
+        if name in mappings:
+            filters['package_version'] = [mappings.get(name).get('version')]
+        elif name in version_constraints:
+            filters[EXTRA_VERSION_CONSTRAINT] = \
+                version_constraints.get(name)
+        return name, filters
+
+    def _resolve_plugin_yaml_url(self, import_url):
+        plugin_spec = import_url.replace(PLUGIN_PREFIX, '', 1).strip()
+        name, plugin_filters = self._make_plugin_filters(
+            plugin_spec, self.version_constraints, self.mappings)
+        plugin = self._find_plugin(name, plugin_filters)
+        if plugin:
+            pass
+            # TODO: fix log
```

Comment:
It's better not to leave TODOs in the code unless they point to a current JIRA.

Reply:
The resolver here has no access to current_app through flask. This log was removed